### PR TITLE
Nix: rework docker image building & uploading (develop)

### PR DESCRIPTION
**Pull request by balsoft**
Companion for #11889

### CODE REVIEW

File: nix/ocaml.nix [hunk 1]
```diff
@@ -188,17 +182,22 @@ let
             src/app/rosetta/rosetta_testnet_signatures.exe \
             src/app/rosetta/rosetta_mainnet_signatures.exe \
             src/app/generate_keypair/generate_keypair.exe \
-            src/app/runtime_genesis_ledger/runtime_genesis_ledger.exe \
-            -j$NIX_BUILD_CORES
+            src/app/archive/archive.exe \
+            src/app/archive_blocks/archive_blocks.exe \
+            src/app/extract_blocks/extract_blocks.exe \
```
Comment (ylecornec) [inline nix/ocaml.nix:187]: Should this be copied in the `installPhase`?

**Review by ylecornec — APPROVED** [review]
Looks good to me, but it looks like it should not be merged before #11838 is resolved right ?
Because the PR for compatible targets `mina_version.runtime`.

**Comment by ylecornec** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**